integration-tests/testutils/report/parser.go

```diff
@@ -40,43 +42,52 @@ var (
 	skipRegexp     = regexp.MustCompile(fmt.Sprintf(commonPattern, "SKIP") + skipPatternSufix)
 )
 
-// ParserReporter is a type implementing io.Writer that
-// parses the input data and sends the results to the Next
-// reporter
+// Statuser reports the status of a test.
+type Statuser interface {
+	Status(subunit.Event) error
+}
+
+// SubunitV2ParserReporter is a type that parses the input data
+// and sends the results as a test status.
 //
 // The input data is expected to be of the form of the textual
 // output of gocheck with verbose mode enabled, and the output
-// will be of the form defined by the subunit format before
-// the binary encoding. There are constants reflecting the
-// expected patterns for this texts.
+// will be of the form defined by the subunit v2 format There
+// are constants reflecting the expected patterns for this texts.
 // Additionally, it doesn't take  into account the SKIPs done
 // from a SetUpTest method, due to the nature of the snappy test
 // suite we are using those for resuming execution after a reboot
 // and they shouldn't be reflected as skipped tests in the final
 // output. For the same reason we use a special marker for the
 // test's announce.
-type ParserReporter struct {
-	Next io.Writer
+type SubunitV2ParserReporter struct {
+	statuser Statuser
```

Comment:
Reading the docs of the io.Writer interface, I think I'll follow your suggestion of splitting parsing and encoding. It says: "Write writes len(p) bytes from p to the underlying data stream", so it shouldn't be transforming the bytes it write on the process. But I'll do that later.